apps/worker/tasks/base.py

```diff
@@ -219,77 +224,116 @@ def apply_async(self, args=None, kwargs=None, **options):
         }
         return super().apply_async(args=args, kwargs=kwargs, headers=headers, **options)
 
-    # Called when attempting to retry the task on db error
-    def _retry(self, countdown=None):
-        if not countdown:
-            countdown = self.default_retry_delay
-
-        try:
-            self.retry(countdown=countdown)
-        except MaxRetriesExceededError:
-            if UploadFlow.has_begun():
-                UploadFlow.log(UploadFlow.UNCAUGHT_RETRY_EXCEPTION)
-            if TestResultsFlow.has_begun():
-                TestResultsFlow.log(TestResultsFlow.UNCAUGHT_RETRY_EXCEPTION)
-
-    def safe_retry(self, max_retries=None, countdown=None, exc=None, **kwargs):
+    def retry(
+        self,
+        countdown=None,
+        exc=None,
+        kwargs=None,
+        max_retries=_NOT_PROVIDED,
+        **other_kwargs,
+    ):
         """
-        Safely retry with max retry limit and proper metrics tracking.
-
-        This method provides a common retry pattern for all tasks with:
-        - Configurable max retry limit
-        - Automatic metric tracking
-        - Consistent error handling
+        Override Celery's retry() to automatically check max_retries and track metrics.
+        If max_retries is not provided, uses self.max_retries from the task class.
+        If max_retries=None is explicitly passed, means unlimited retries (matches Celery behavior).
 
         Args:
-            max_retries: Maximum number of retries allowed (default: task's max_retries)
-            countdown: Seconds to wait before retry (default: exponential backoff)
+            countdown: Seconds to wait before retry (optional)
             exc: Exception to include in retry (optional)
-            **kwargs: Additional kwargs to pass to self.retry()
-
-        Returns:
-            True if retry was scheduled
-            False if max retries exceeded
-
-        Example:
-            if some_condition_requires_retry:
-                if not self.safe_retry(max_retries=5, countdown=60):
-                    # Max retries exceeded
-                    log.error("Giving up after too many retries")
-                    return {"success": False, "reason": "max_retries"}
+            kwargs: Task kwargs to use when retrying (optional, Celery-specific parameter)
+            max_retries: Maximum number of retries allowed (default: task's max_retries).
+                        Pass None explicitly for unlimited retries (matches Celery behavior).
+            **other_kwargs: Additional kwargs to pass to Celery's retry()
         """
-        current_retries = self.request.retries if hasattr(self, "request") else 0
-        task_max_retries = max_retries if max_retries is not None else self.max_retries
-
+        request = getattr(self, "request", None)
```

Comment:
can this just return out if `self.request` is `None`? The next line is going to freak out trying to call `retries` on `None`